src/consent-manager-builder/index.js

```diff
@@ -75,9 +77,17 @@ export default class ConsentManagerBuilder extends Component {
     })
   }
 
-  componentDidMount() {
-    // TODO: handle errors properly
-    this.initialise()
+  async componentDidMount() {
+    const {onError} = this.props
+    if (onError && typeof onError === 'function') {
+      try {
+        await this.initialise()
+      } catch (e) {
+        await onError(e)
```

Comment:
Do we need to await `onError`? Seems like this could be a fire-and-forget.

Reply:
Could be, I followed netto's suggestion here https://github.com/segmentio/consent-manager/pull/31#discussion_r272323848

Reply:
Ship away